test(bundle): periodic e2e tests for the operator bundle

SteveMattar opened the pull request:
Depends on submariner-io/shipyard/pull/599

Closes #1369

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

skitt commented on .github/workflows/periodic.yml:
```diff
@@ -26,3 +26,28 @@ jobs:
           title: Broken link detected by CI
           content-filepath: .github/ISSUE_TEMPLATE/broken-link.md
           labels: automated, broken link
+  test-bundle-periodic:
```
Please add a blank line above this new entry.

skitt commented on .github/workflows/periodic.yml:
```diff
@@ -26,3 +26,28 @@ jobs:
           content-filepath: .github/ISSUE_TEMPLATE/broken-link.md
           labels: automated, broken link
+  test-bundle-periodic:
+    name: Markdown Links (all files)
```
```suggestion
    name: Operator Bundle Test
```

skitt commented on package/Dockerfile.submariner-operator-index:
```diff
@@ -0,0 +1,14 @@
+FROM quay.io/operator-framework/upstream-registry-builder as builder
+
+COPY packagemanifests manifests
+RUN  ls -R manifests/**/*.clusterserviceversion.yaml | xargs -I% sed -i 's/replaces: submariner.v0.0.0//g' % && \
```
```suggestion
RUN find manifests -name \*.clusterserviceversion.yaml -exec sed -i 's/replaces: submariner.v0.0.0//g' {} + && \
```

dfarrell07 commented on Makefile:
```diff
@@ -183,12 +192,13 @@ is-semantic-version:
-KUSTOMIZE := $(CURDIR)/bin/kustomize
-$(KUSTOMIZE): vendor/modules.txt
+# TODO: a workaround until this issue will be fixed https://github.com/kubernetes-sigs/kustomize/issues/4008
+$(KUSTOMIZE):
```
Doesn't this introduce a type of supply-chain vulnerability like @skitt fixed in f71061f5fec92b1308e7005285710c5cfc30b006? We might need to do those verifications here.

SteveMattar replied:
I did not find the checksum on the source repo. I will try to look elsewhere.

dfarrell07 replied:
This might be something that's better documented in an Issue for now and fixed here (and elsewhere?) next sprint.

SteveMattar replied:
Fixed, I found the `checksum.txt` file on github. Thanks for the notice @dfarrell07

dfarrell07 commented on .github/ISSUE_TEMPLATE/broken-bundle.md:
```diff
@@ -0,0 +1,4 @@
+# Broken bundle
+
+Periodic operator bundle tests CI detected a broken bundle. Please see the [periodic job
```
```suggestion
Periodic operator bundle CI detected a broken bundle. Please see the [periodic job
```

dfarrell07 commented on .github/workflows/periodic.yml:
```diff
@@ -26,3 +26,29 @@ jobs:
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v2
+      - name: Fetch all git tags
```
```suggestion

      - name: Fetch all git tags
```

Maybe blanks between these sections?

dfarrell07 commented on .github/workflows/periodic.yml:
```diff
@@ -26,3 +26,29 @@ jobs:
+        uses: actions/checkout@v2
+      - name: Fetch all git tags
+        run: git fetch origin +refs/tags/*:refs/tags/*
+      - name: Generate submariner version
```
```suggestion
      - name: Generate Submariner version
```

dfarrell07 commented on .github/workflows/periodic.yml:
```diff
@@ -26,3 +26,29 @@ jobs:
+          echo "VERSION=${VERSION}" >> $GITHUB_ENV
+          echo "FROM_VERSION=${FROM_VERSION}" >> $GITHUB_ENV
+          echo "CHANNEL=alpha-$(echo ${VERSION} | cut -d'.' -f1,2)" >> $GITHUB_ENV
+      - name: Run bundle test
```
```suggestion
      - name: Run E2E with using the bundle deployment method
```

Commit:
fix(kustomize): download kustomize from github releases

This is a workaround until the issue will be fixed
kubernetes-sigs/kustomize#4008.

Signed-off-by: Steve Mattar <smattar@redhat.com>

Commit:
test(bundle): periodic e2e tests for the operator bundle

Depends on submariner-io/shipyard/pull/599.
Closes #1369.

Signed-off-by: Steve Mattar <smattar@redhat.com>

dfarrell07 commented on .github/workflows/periodic.yml:
```diff
@@ -26,3 +26,38 @@ jobs:
+          echo "REPO=localhost:5000" >> $GITHUB_ENV
+          echo "IMG=${REPO}/submariner-operator:local" >> $GITHUB_ENV
+
+      - name: Run E2E with using the bundle deployment method
```
```suggestion
      - name: Run E2E using the bundle deployment method
```

Sorry my suggestion needed a suggestion :sweat_smile:

dfarrell07 commented on .github/workflows/periodic.yml:
```diff
@@ -26,3 +26,40 @@ jobs:
           labels: automated, broken link
+
+  test-bundle-periodic:
+    name: Operator Bundle Test
```
Since there's some coverage for this in Shipyard's scripts tests now, IMHO it might make sense for this to be a Flake Finder job (moved to that GHA workflow).

skitt committed:
Merge branch 'devel' into test-bundle
